### PR TITLE
[fix]: Validate name field on change

### PR DESCRIPTION
**Pull request by deeonwuli**
### :pushpin: References

-   **Issue:** Closes https://app.clickup.com/t/8694qft5g

### :memo: Implementation
- [x] Add compulsory `shortName` field
- [x] Validate name field to ensure predictor name is unique

### :art: Screenshots

https://github.com/user-attachments/assets/19a0e3a2-08eb-465b-bb69-faf377d9484e

### :fire: Testing

#8694qft5g

### CODE REVIEW

**Comment by ifoche** [comment]
Task linked: [CU-8694qft5g better feedback for errors](https://app.clickup.com/t/8694qft5g)

File: src/webapp/components/predictor-form/PredictorForm.tsx [hunk 1]
```diff
@@ -64,6 +76,16 @@ const useValidations = (
...
         case "id":
             return { validation: createPattern(fullUidRegex, i18n.t("Please provide a valid identifier")) };
+        case "name":
+            return {
+                validation: value => {
+                    const nameValueExists = existingPredictors.includes(value)
+                        ? i18n.t("Name must be unique")
+                        : undefined;
+
+                    return hasValue(value) || nameValueExists;
```
Comment (gqcorneby) [inline src/webapp/components/predictor-form/PredictorForm.tsx]: just a minor optimization but good for scalability. We can short-circuit the validation by doing an early return if it fails the hasValue check. Something like   
```
const requiredError = hasValue(value);
if (requiredError) return requiredError;
return existingPredictors.includes(value) ? i18n.t("Name must be unique") : undefined;
```

**Review by gqcorneby — COMMENTED** [review]
Thank you @deeonwuli! Just a comment for minor optimization on the validation. 

Do we also want to check uniqueness on the shortName? I tried testing duplicate shortName and I got a non-descriptive error in the form.
image

This was the error `Property `shortName` with value `test` on object test2 [eohrk12yeRV] (Predictor) already exists on object KhGbuqNgPcJ`

File: src/webapp/components/predictor-form/PredictorForm.tsx (hunk 1 above)
Comment (deeonwuli) [reply src/webapp/components/predictor-form/PredictorForm.tsx]: good catch @gqcorneby 👌

File: src/webapp/components/predictor-edit-wizard/PredictorEditWizard.tsx [hunk 2]
```diff
@@ -111,7 +119,7 @@ export const PredictorEditWizard: React.FC<PredictorEditWizardProps> = ({ predic
     );
 };
 
-const Wizard: React.FC<{ onCancel: any }> = ({ children, onCancel }) => {
+const Wizard: React.FC<{ onCancel: any; steps: WizardStep[] }> = ({ children, onCancel, steps }) => {
```
Comment (gqcorneby) [inline src/webapp/components/predictor-edit-wizard/PredictorEditWizard.tsx:122]: This isn't yours but we can update any. I think it would be () => void

File: src/webapp/components/predictor-form/PredictorForm.tsx [hunk 3]
```diff
@@ -23,21 +23,46 @@ import { OutputFF } from "./components/OutputFF";
...
 
     const [expressionValidation, setExpressionValidation] = React.useState<
         Record<string, ExpressionValidation | undefined>
     >({});
+    const [existingPredictors, setExistingPredictors] = React.useState<PredictorDetails[]>([]);
+
+    useEffect(() => {
+        compositionRoot.predictors.list().run(
+            ({ objects: predictors }) =>
+                setExistingPredictors(predictors.filter(existingPredictor => existingPredictor.id !== predictor?.id)),
```
Comment (gqcorneby) [inline src/webapp/components/predictor-form/PredictorForm.tsx:50]: :+1:

**Review by gqcorneby — APPROVED** [review]
Working great, @deeonwuli! just a non-blocking comment
